src/notary/src/App.js

```diff
@@ -1,112 +1,151 @@
+                    state.forms.registerMachine._disabled = false;
+                    this.setState(state);
+                })
+        })
+    }
+
+    callCreateMachine() {
+        const MachineOwner = contract(MachineOwnerContract);
+        MachineOwner.setProvider(this.state.web3.currentProvider);
+
+        const address = this.state.contracts.machineOwner.address;
+        let _, model, make, vin;
+        ({ _, model, make, vin } = this.state.forms.registerMachine);
+
+        this.state.web3.eth.getAccounts((error, accounts) => {
+            MachineOwner.at(address).then((instance) => {
+                instance.createNewMachine(model, make, vin, { from: accounts[0] })
+                    .then((result) => {
+                        const state = { ...this.state };
+                        const machine = {
+                            model: model,
+                            make: make,
+                            vin: vin,
+                            address: result.logs[0].args.newMachine
+                        };
+                        state.contracts.machineOwner.machines.push(machine);
+                        this.setState(state);
+                    });
+            })
+        })
+    };
+
+    machineFormChangedHandler = (e) => {
+        const name = e.target.name;
+        const value = e.target.value;
+
+        // Update form
+        const form = { ...this.state.forms.registerMachine };
+        form[name] = value;
+
+        // Validate form input
+        const { _disabled, model, make, vin } = form;
+        const isValid = make !== '' && model !== '' && vin !== '';
+        form._disabled = !isValid;
+
+        // Update state
+        const state = { ...this.state };
+        state.forms.registerMachine = form;
+        this.setState(state);
+    };
+
+    machineFormSubmittedHandler = (e) => {
+        e.preventDefault();
+        if (!this.state.forms.registerMachine._disabled) {
+            this.callCreateMachine()
+        }
+    };
+
+    ownerFormSubmittedHandler = (e) => {
+        e.preventDefault();
+        if (!this.state.forms.registerOwner._disabled) {
+            this.instantiateMachineOwnerContract()
+        }
+    };
+
+    render() {
+        return (
+            <div className="App">
+                <div className="uk-container uk-container-center uk-margin-top uk-margin-large-bottom">
+                    <div className="uk-grid-match" data-uk-grid-margin>
+                        <RegisterOwnerForm
+                            onSubmit={(e) => this.ownerFormSubmittedHandler(e)}
+                            state={this.state.forms.registerOwner}
+                        />
+                        <RegisterMachineForm
+                            onChange={(e) => this.machineFormChangedHandler(e)}
+                            onSubmit={(e) => this.machineFormSubmittedHandler(e)}
+                            state={this.state.forms.registerMachine}
```

Comment:
you refer to `this.state` more than once, it would be worth it to deconstruct it at the top of the function
```
const { state } = this;
```
or
```
const { forms, contracts } = this.state;
```